app/controllers/application_controller.rb

```diff
@@ -14,9 +15,25 @@ def require_current_user
     redirect_to root_path
   end
 
-  def generate_flash(resource)
-    resource.errors.messages.each do |validation, message|
-      flash[validation] = "#{validation}: #{message}"
+  def current_admin?
+    current_user && current_user.admin?
+  end
+
+  def require_admin
+    render file: 'public/404', status: :not_found unless current_admin?
+  end
+
+  def exclude_admin
+    render file: 'public/404', status: :not_found if current_admin?
+  end
+
+  def destroy_unsaved_npcs
+    NpcModel.where(is_saved: 0).ids.map do |id|
+      NpcModelArmor.where(npc_model_id: id).destroy_all
+      NpcModelPack.where(npc_model_id: id).destroy_all
+      NpcModelWeapon.where(npc_model_id: id).destroy_all
+      NpcModelSpell.where(npc_model_id: id).destroy_all
     end
+    NpcModel.where(is_saved: 0).destroy_all
```

Comment:
I moved this logic here to be a helper method, but I took it out from all the other pages except the welcome page, so it could have just stayed there, but maybe we'll decide to add it someplace else too ... and this cleans up that index method on the `welcome_controller` too.